nix/ec2.nix

```diff
@@ -358,7 +358,21 @@ in
       example = { "/dev/xvdb".disk = "ephemeral0"; "/dev/xvdg".disk = "vol-00000000"; };
       type = with types; attrsOf (submodule ec2DiskOptions);
       description = ''
-        Block device mapping.  <filename>/dev/xvd[a-e]</filename> must be ephemeral devices.
+        Block device mapping.
+
+        <filename>/dev/sd[a-e]</filename> or <filename>/dev/xvd[a-e]</filename> must be ephemeral devices.
+
+        nvme devices should have name like <filename>/dev/nvme[1-26]n1</filename>, the number in device name should not be skipped.
```

Comment:
Can you clarify this sentence a bit?  It's not immediately clear what is meant with it.

Reply:
```
With the following instances, EBS volumes are exposed as NVMe block devices: C5, C5d, i3.metal, M5, and M5d (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html). For this instances volumes should be attached as <filename>/dev/nvme[1-26]n1</filename>, the number in device name should not be skipped.
```

@rbvermaa

Reply:
@rbvermaa is the refactored sentence above better? Can I make a commit?

Reply:
@srghma With skipped, do you mean that there should be no 'hole' in the numbering of the devices? If so, are you only allowed to detach the volumes in 'descending' order, aka only the last one each time.

Also, in the above sentence 'For this instances' should be 'For these instance'.

Reply:
@rbvermaa 
1. yes, there should be no hole in numbering, will change
2. 

> detach the volumes in 'descending' order

I think user can deattach any volume he wants

lets consider an example

```
{
  machine =
    { resources, pkgs, ... }:
    {
      fileSystems."/disk1" = {
        autoFormat = true;
        fsType = "ext4";
        device = "/dev/nvme1n1";
        ec2.disk = resources.ebsVolumes.disk1;
      };

      # user want to deattach it
      fileSystems."/disk2" = {
        autoFormat = true;
        fsType = "ext4";
        device = "/dev/nvme2n1";
        ec2.disk = resources.ebsVolumes.disk2;
      };

      fileSystems."/disk3" = {
        autoFormat = true;
        fsTye = "ext4";
        device = "/dev/nvme3n1";
        ec2.disk = resources.ebsVolumes.disk3;
      };
    };
}

```

if use just comments second device

```
{
  machine =
    { resources, pkgs, ... }:
    {
      fileSystems."/disk1" = {
        autoFormat = true;
        fsType = "ext4";
        device = "/dev/nvme1n1";
        ec2.disk = resources.ebsVolumes.disk1;
      };

      # user want to deattach it
      # fileSystems."/disk2" = {
      #   autoFormat = true;
      #   fsType = "ext4";
      #   device = "/dev/nvme2n1";
      #   ec2.disk = resources.ebsVolumes.disk2;
      # };

      fileSystems."/disk3" = {
        autoFormat = true;
        fsTye = "ext4";
        device = "/dev/nvme3n1"; # will be attached as /dev/nvme2n1 and will cause mount error "cant find device ..."
        ec2.disk = resources.ebsVolumes.disk3;
      };
    };
}

```

right solution is 


```
{
  machine =
    { resources, pkgs, ... }:
    {
      fileSystems."/disk1" = {
        autoFormat = true;
        fsType = "ext4";
        device = "/dev/nvme1n1";
        ec2.disk = resources.ebsVolumes.disk1;
      };

      # user want to deattach it
      # fileSystems."/disk2" = {
      #   autoFormat = true;
      #   fsType = "ext4";
      #   device = "/dev/nvme2n1";
      #   ec2.disk = resources.ebsVolumes.disk2;
      # };

      fileSystems."/disk3" = {
        autoFormat = true;
        fsTye = "ext4";
        device = "/dev/nvme2n1"; # will be attached as /dev/nvme2n1
        ec2.disk = resources.ebsVolumes.disk3;
      };
    };
}

```